authors/apps/profiles/views.py

```diff
@@ -140,3 +140,35 @@ def get(self, request):
             return Response(data=msg, status=status.HTTP_400_BAD_REQUEST)
         msg = {'following': following}
         return Response(data=msg, status=status.HTTP_200_OK)
+
+
+class ReadingStatsView(generics.ListAPIView):
+    """View class for retrieving the reading statistics of a user"""
+    permission_classes = (IsAuthenticated, )
+    serializer_class = ReadingStatsSerializer
+    renderer_classes = (ReadingStatsJSONRenderer, )
+
+    def get_queryset(self, *args, **kwargs):
+        user = self.kwargs.get('username')
+        records = ReadingStat.objects.all().filter(user=user)
+
+        return [articles.articles for articles in records]
+
+    def get(self, request, username):
+
+        if username != request.user.username:
+            raise PermissionDenied({
+                "message": "Unauthorized to view these stats"
+            })
+        articles = self.get_queryset()
+        total_read_time = 0
+        for article in articles:
+            total_read_time += article.read_time
+
+        data = {
+            'user': request.user.username,
+            'no._of_articles_read': len(articles),
+            'total_read_time': total_read_time,
+            'recent_articles': [article.slug for article in articles[:5]],
```

Comment:
@Opio-Emmanuel-Omona Could we return the id because the whenever an article is edited the slug changes thus an article may not be trackable from this list. What do you think??
```suggestion
            'recent_articles': [article.id for article in articles[:5]],
```

Reply:
@archbishop12 Yes, I think that is a good idea since the slug changes.
I have made the change.